bindings/java/src/main/java/org/ucx/jucx/Worker.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+package org.ucx.jucx;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * Worker is the object representing a local communication resource such as a
+ * network interface or host channel adapter port.
+ */
+@SuppressWarnings("unused")
+public class Worker implements Closeable {
+    public static final int MAX_QUEUED_EVENTS = (1 << 20);
+
+    private long            nativeId;
+    private CompletionQueue compQueue;
+    private byte[]          workerAddress;
+    private Callback        callback;
+    private int             maxEvents;
+    private boolean         closed;
+
+    /**
+     * Creates a new Worker.
+     *
+     * @param cb
+     *            Implementation of Worker.Callback interface
+     *
+     * @param maxEvents
+     *            Number of maximum queued completed events
+     *
+     * @throws IllegalArgumentException
+     *             In case cb == null or maxEvents <= 0 or
+     *             maxEvents > MAX_QUEUED_EVENTS
+     *
+     * @throws UnsatisfiedLinkError
+     *             In case an error while loading native libraries
+     *
+     * @throws IOException
+     *             In case native Worker creation failed
+     */
+    public Worker(Callback cb, int maxEvents) throws IOException {
+        if (cb == null || maxEvents <= 0 || maxEvents > MAX_QUEUED_EVENTS) {
+            throw new IllegalArgumentException();
+        }
+
+        // An error occurred while loading native libraries
+        if (LoadLibrary.errorMessage != null) {
+            throw new UnsatisfiedLinkError(LoadLibrary.errorMessage);
+        }
+
+        this.callback       = cb;
+        this.closed         = false;
+        this.maxEvents      = maxEvents;
+        this.workerAddress  = null;
+        this.compQueue      = new CompletionQueue(); // Shared queue wrapper
+        this.nativeId       = Bridge.createWorker(maxEvents, compQueue, this);
+        if (nativeId == 0) {
+            throw new IOException("Failed to create Worker");
+        }
+
+        // Setting shared queue endianness to be the same as native (C) code
+        this.compQueue.setEndianness();
+    }
+
+    /**
+     * Getter for native pointer as long.
+     *
+     * @return long integer representing native pointer
+     */
+    public long getNativeId() {
```

Comment:
why is this public? 
any reason for an application to access this? (I see this in the unit test)

shouldn't it be protected? so that class bridge can access it but not applications